lib/request.js

```diff
@@ -67,6 +67,7 @@ function Request (options) {
   this.connection = {
     remoteAddress: options.remoteAddress || '127.0.0.1'
   }
+  this.socket = this.connection
```

Comment:
Should there be some sort of deprecation here? We'd have to follow the upstream deprecation cycle.

Reply:
I think you can add the deprecate warning in the [typing](https://github.com/fastify/light-my-request/blob/master/index.d.ts#L63) to something like this
```typescript
interface Request extends stream.Readable {
...
// @deprecated
connection: object
...
```

Reply:
I agree with @jsumners.
According to the [Node.js docs](https://nodejs.org/api/http.html#http_request_connection), `.connection` has been deprecated in favor of `.socket`.
I think we should do something like this:
```js
this.socket = {
  remoteAddress: options.remoteAddress || '127.0.0.1'
}

Object.defineProperty(Request, 'connection', {
  get () {
    // if node major ≥13 emit a warning with https://github.com/fastify/fastify-warning
    return this.socket
  }
})
```

Reply:
+1!